### PR TITLE
fix: rename "phone" courier channel to "sms"

### PR DESCRIPTION
**Pull request by jonas-jonas**
## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [ ] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got the approval (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](https://github.com/ory/docs).

### CODE REVIEW

**Review by aeneasr — APPROVED** [review]

**Review by hperl — DISMISSED** [review]

File: courier/sms.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ func (c *courier) QueueSMS(ctx context.Context, t SMSTemplate) (uuid.UUID, error
 	message := &Message{
 		Status:       MessageStatusQueued,
 		Type:         MessageTypeSMS,
-		Channel:      "sms",
+		Channel:      "phone",
```
Comment (zepatrik) [inline courier/sms.go:32]: Why use `"phone"`? IMO, this is not ideal as it is not clear whether the message is delivered as an SMS or phone call.

**Review by zepatrik — CHANGES_REQUESTED** [review]

File: courier/sms.go (hunk 1 above)
Comment (aeneasr) [reply courier/sms.go:32]: I kinda agree with this, see discussion here: https://github.com/ory/kratos/pull/3649#discussion_r1436955299

Comment (zepatrik) [reply courier/sms.go:32]: So you agree with sms or phone? I just thought that it makes more sense to make this as precise as possible, because it would be confusing to have later "phone" and "phone_call".

In the end no one forces you to use this for sms, but that's why we have the channels. One can add "phone_call", "whatsapp", "my_native_app_push_notification", or any other channel (once we fully expose that).

I am also not sure if we really need to mirror all the intermediary states of phone calls, all that Kratos cares about is whether the correct code was submitted, regardless how the user got the code.

Comment (aeneasr) [reply courier/sms.go:32]: I thought that `sms` made the most sense

Comment (jonas-jonas) [reply courier/sms.go]: Okay, I refactored the whole thing to use "sms" now.

File: identity/extension_verification.go [hunk 2]
```diff
@@ -24,40 +32,58 @@ func NewSchemaExtensionVerification(i *Identity, lifespan time.Duration) *Schema
...
+			formatString = "email"
+			formatter, ok := jsonschema.Formats[formatString]
+			if !ok {
+				supportedKeys := maps.Keys(jsonschema.Formats)
+				return ctx.Error("format", "format %q is not supported. Supported formats are [%s]", formatString, strings.Join(supportedKeys, ", "))
+			}
+
+			if !formatter(value) {
+				return ctx.Error("format", "%q is not valid %q", value, formatString)
+			}
```
Comment (zepatrik) [inline identity/extension_verification.go:70]: This block needs to be moved to after the if, right?

File: identity/identity_verification.go [hunk 3]
```diff
@@ -57,7 +57,7 @@ type VerifiableAddress struct {
 	// enum: ["email"]
```
Comment (zepatrik) [inline identity/identity_verification.go:57]:
```suggestion
	// enum: ["email", "sms"]
```
or remove this.
The generated SDK probably still respects this comment.

File: identity/extension_verification.go [hunk 4]
```diff
@@ -24,40 +32,58 @@ func NewSchemaExtensionVerification(i *Identity, lifespan time.Duration) *Schema
 	return &SchemaExtensionVerification{i: i, lifespan: lifespan}
 }
 
+const (
+	ChannelTypeEmail = "email"
+	ChannelTypeSMS   = "sms"
+	ChannelTypePhone = "phone"
```
Comment (zepatrik) [inline identity/extension_verification.go:38]:
```suggestion
```

File: identity/identity_verification.go [hunk 5]
```diff
@@ -25,7 +25,7 @@ const (
 // VerifiableAddressType must not exceed 16 characters as that is the limitation in the SQL Schema
 //
 // swagger:model identityVerifiableAddressType
-type VerifiableAddressType string
+type VerifiableAddressType = string
```
Comment (zepatrik) [inline identity/identity_verification.go:28]: Can we remove this type completely? Or should we keep it for documentation purposes? If yes, I'd suggest to revert the changes where you replace this type by string in other places.

**Review by zepatrik — COMMENTED** [review]
Just some minor things. For recovery we'll do the same, but we don't yet support SMS recovery, right?

File: identity/extension_verification.go (hunk 2 above)
Comment (zepatrik) [reply identity/extension_verification.go:70]: Ah I guess not, because the library already did the check and we don't want to double-validate.

File: identity/identity_verification.go (hunk 5 above)
Comment (jonas-jonas) [reply identity/identity_verification.go:28]: It's still used in a lot of places. I think keeping it doesn't hurt.

File: identity/extension_verification.go (hunk 2 above)
Comment (jonas-jonas) [reply identity/extension_verification.go:70]: yup, exactly, the library doesn't de-duplicate the error messages, so this would lead to ugly errors.

**Review by zepatrik — APPROVED** [review]
